lib/pos_blockchain.go

```diff
@@ -646,6 +633,19 @@ func (bc *Blockchain) validateAndIndexBlockPoS(block *MsgDeSoBlock) (*BlockNode,
 		return blockNode, nil
 	}
 
+	// Base case - Check if the block has already been stored and fails the timestamp drift check.
+	// If it fails the check, then we leave it as stored and return early.
+	if exists && blockNode.IsStored() {
+		// If the block is too far in the future, we leave it as STORED and return early.
+		failsTimestampDriftCheck, err := bc.isBlockTimestampTooFarInFuturePoS(block.Header)
```

Comment:
When check the timestamp drift here as a base case to prevent recursive validation calls on the block's ancestors. If the check fails, then we can leave the block as STORED and exit early